Sources/Commandant/Command.swift

```diff
@@ -205,10 +205,29 @@ extension CommandRegistry {
 
 		func launchTask(_ path: String, arguments: [String]) -> Int32 {
 			let task = Process()
-			task.launchPath = path
 			task.arguments = arguments
 
-			task.launch()
+			do {
+			#if canImport(Darwin)
+				if #available(macOS 10.13, *) {
+					task.executableURL = URL(fileURLWithPath: path)
+					try task.run()
+				} else {
+					task.launchPath = path
+					task.launch()
+				}
+			#elseif compiler(>=5)
+				task.executableURL = URL(fileURLWithPath: path)
+				try task.run()
+			#else
+				task.launchPath = path
+				task.launch()
+			#endif
+			} catch let nserror as NSError {
+				return Int32(truncatingIfNeeded: nserror.code)
+			} catch {
+				return -1
```

Comment:
Should these return `task.terminationStatus`?

Reply:
I believe that this code block will never be executed.🤔
Should I replace it with `fatalError()` or something?

Reply:
I think both `catch`s could be combined and return `task.terminationStatus`?

Reply:
`task.run()` may throws `NSError` without executing.
https://github.com/apple/swift-corelibs-foundation/blob/master/Foundation/Process.swift#L470-L484
On such conditions, `task.terminationStatus` is `0`, so we need to catch `NSError`.

Reply:
> On such conditions, `task.terminationStatus` is `0`

On such conditions, accessing `task.terminationStatus` produces `NSInvalidArgumentException`
```terminal.sh-session
$ swift
swift_oss_helper command enabled.
Welcome to Apple Swift version 4.2.1 (swiftlang-1000.11.42 clang-1000.11.45.1). Type :help for assistance.
  1> import Foundation
  2> let task = Process()
task: Process = {
  ObjectiveC.NSObject = {
    baseNSObject@0 = {
      isa = NSConcreteTask
    }
  }
}
  3> task.executableURL = URL(fileURLWithPath: "/dev/null")
  4> do { 
  5.     try task.run() 
  6. } catch { 
  7.     print(task.terminationStatus) 
  8. } 
2019-03-12 09:10:33.932363+0900 repl_swift[3638:58597] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** -[NSConcreteTask terminationStatus]: task not launched'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff3a23fd49 __exceptionPreprocess + 256
	1   libobjc.A.dylib                     0x00007fff64dbda17 objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff3a23fb7b +[NSException raise:format:] + 201
	3   Foundation                          0x00007fff3c427ca3 -[NSConcreteTask terminationStatus] + 104
	4   ???                                 0x00000001000fc28a 0x0 + 4296000138
	5   repl_swift                          0x0000000100000ea0 main + 0
	6   libdyld.dylib                       0x00007fff665ea3d5 start + 1
)
```

Reply:
Thanks for checking! The documentation for `Process` is not good.